dotnet/src/Connectors/Connectors.Memory.Weaviate/WeaviateVectorStore.cs

```diff
@@ -0,0 +1,90 @@
+/// </summary>
+/// <remarks>
+/// This class can be used with collections of any schema type, but requires you to provide schema information when getting a collection.
+/// </remarks>
+public sealed class WeaviateVectorStore : IVectorStore
+{
+    /// <summary><see cref="HttpClient"/> that is used to interact with Weaviate API.</summary>
+    private readonly HttpClient _httpClient;
+
+    /// <summary>Optional configuration options for this class.</summary>
+    private readonly WeaviateVectorStoreOptions _options;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="WeaviateVectorStore"/> class.
+    /// </summary>
+    /// <param name="httpClient"><see cref="HttpClient"/> that is used to interact with Weaviate API.</param>
+    /// <param name="options">Optional configuration options for this class.</param>
+    public WeaviateVectorStore(HttpClient httpClient, WeaviateVectorStoreOptions? options = null)
+    {
+        Verify.NotNull(httpClient);
+
+        this._httpClient = httpClient;
+        this._options = options ?? new();
+    }
+
+    /// <inheritdoc />
+    public IVectorStoreRecordCollection<TKey, TRecord> GetCollection<TKey, TRecord>(string name, VectorStoreRecordDefinition? vectorStoreRecordDefinition = null)
+        where TKey : notnull
+        where TRecord : class
+    {
+        if (typeof(TKey) != typeof(Guid))
+        {
+            throw new NotSupportedException($"Only {nameof(Guid)} key is supported.");
+        }
+
+        if (this._options.VectorStoreCollectionFactory is not null)
+        {
+            return this._options.VectorStoreCollectionFactory.CreateVectorStoreRecordCollection<TKey, TRecord>(
+                this._httpClient,
+                name,
+                vectorStoreRecordDefinition);
+        }
+
+        var recordCollection = new WeaviateVectorStoreRecordCollection<TRecord>(
+            this._httpClient,
+            name,
+            new()
+            {
+                VectorStoreRecordDefinition = vectorStoreRecordDefinition
+            }) as IVectorStoreRecordCollection<TKey, TRecord>;
+
+        return recordCollection!;
+    }
+
+    /// <inheritdoc />
+    public async IAsyncEnumerable<string> ListCollectionNamesAsync([EnumeratorCancellation] CancellationToken cancellationToken = default)
+    {
+        using var request = new WeaviateGetCollectionsRequest().Build();
+
+        var response = await this._httpClient.SendWithSuccessCheckAsync(request, cancellationToken).ConfigureAwait(false);
+        var responseContent = await response.Content.ReadAsStringWithExceptionMappingAsync().ConfigureAwait(false);
+        var collectionResponse = JsonSerializer.Deserialize<WeaviateGetCollectionsResponse>(responseContent);
+
+        if (collectionResponse?.Collections is not null)
+        {
+            foreach (var collection in collectionResponse.Collections)
+            {
+                if (!string.IsNullOrWhiteSpace(collection.CollectionName))
```

Comment:
Can this happen and if yes is this expected or an error state?

The behaviour here is to ignore the invalid collection name, I dont think we should make this decision for the host application, consider either passing the collection name to the next level as-is or throwing.

Reply:
I don't expect this to happen, so I made `CollectionName` non-nullable, so the method should always return some name.